modules/fedimint-mint-client/src/client_db.rs

```diff
@@ -128,7 +128,9 @@ pub(crate) fn migrate_state_to_v2(
 
     let new_mint_state_machine = match mint_client_state_machine_variant {
         0 => {
-            let old_state = MintOutputStateMachineV1::consensus_decode(cursor, &decoders)?;
+            let _output_sm_len = u16::consensus_decode(cursor, &decoders)?;
```

Comment:
We're failing to decode the structs in the migration if we don't first read the length.

Reply:
Makes sense. I always forget about this extra len in variants.

But, I copied the code here from `modules/fedimint-dummy-client/src/db.rs` `get_v1_migrated_state` and I don't see it decoding any len either, so does it mean it's broken as well, and just was never actually tested?

Reply:
Hmm we do test it in `test_client_db_migrations` of the dummy tests crate, but perhaps its decoding to the wrong values. We don't test that part. I'll check

Reply:
We can address the dummy client migrations in a followup. Right now this is making it difficult to continue triaging the rest of the failures in the upgrade tests, so if there isn't a logical error let's land soon :slightly_smiling_face:

Reply:
All the reason why the dummy migration tests work is just a happy accident lol. The `Unreachable` state decodes a `u16` and I guess originally I thought this was the `ModuleInstanceId`, but its actually the length. 

```
impl Decodable for Unreachable {
    fn consensus_decode<R: std::io::Read>(
        reader: &mut R,
        modules: &ModuleDecoderRegistry,
    ) -> Result<Self, fedimint_core::encoding::DecodeError> {
        let module_instance_id = ModuleInstanceId::consensus_decode(reader, modules)?;
        let operation_id = OperationId::consensus_decode(reader, modules)?;
        let txid = TransactionId::consensus_decode(reader, modules)?;
        let amount = Amount::consensus_decode(reader, modules)?;

        Ok(Unreachable {
            _module_instance_id: module_instance_id,
            operation_id,
            txid,
            amount,
        })
    }
}
```

I can submit a quick fix.

Reply:
https://github.com/fedimint/fedimint/pull/6362